contracts/consensus/Consensus.sol

```diff
@@ -374,4 +353,61 @@ contract Consensus {
 
         proposals[_proposal] = committee_;
     }
+
+    function verifyCommitteeLock(
+        bytes20 _chainId,
+        bytes32 _kernelHash,
+        bytes32 _originObservation,
+        uint256 _dynasty,
+        uint256 _accumulatedGas,
+        bytes32 _committeeLock,
+        bytes32 _source,
+        bytes32 _target,
+        uint256 _sourceBlockHeight,
+        uint256 _targetBlockHeight
+    )
+        private
+        view
+    {
+        address coreAddress = assignments[_chainId];
+        require(
+            isCore(coreAddress),
+            "There is no core for the specified chain id"
+        );
+
+        bytes32 transitionHash = Core(coreAddress).hashTransition(
```

Comment:
can be one function in `Core.assertProposal(KernelHash, T, VM) returns (bytes32 _proposal)` and requires `KernelHash` equals `OpenKernelHash` (and is precommitted) (this is redundantly checked then...)